.github/workflows/test.yml

```
@@ -24,6 +29,9 @@ jobs:
       - uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.version }}
-          cache: 'pnpm'
-      - run: pnpm install
+          cache: ${{ !inputs.react_version && 'pnpm' || '' }}
+      - if: ${{ inputs.react_version }}
+        run: |
+          pnpm -r update react@${{ inputs.react_version }} react-dom@${{ inputs.react_version }} react-server-dom-webpack@${{ inputs.react_version }}
+      - run: pnpm install --frozen-lockfile
```

We decided to remove it as it's the default behavior.
```suggestion
      - run: pnpm install
```